src/pybind/mgr/dashboard/controllers/nvmeof.py

```diff
@@ -66,13 +98,28 @@ def get(self, nqn: str):
         )
         @empty_response
         @handle_nvmeof_error
-        def create(self, nqn: str, enable_ha: bool, max_namespaces: int = 256):
-            return NVMeoFClient().stub.create_subsystem(
+        def create(self, nqn: str, enable_ha: bool,
+                   max_namespaces: int = 256, initiators: str = None):
+            subsystem_response = NVMeoFClient().stub.create_subsystem(
                 NVMeoFClient.pb2.create_subsystem_req(
                     subsystem_nqn=nqn, max_namespaces=max_namespaces, enable_ha=enable_ha
                 )
             )
 
+            if subsystem_response.status != 0:
+                return subsystem_response
+
+            if initiators is not None:
+                all_host_nqns = initiators.split(',')
+                for host_nqn in all_host_nqns:
+                    add_host_response = NVMeoFClient().stub.add_host(
+                        NVMeoFClient.pb2.add_host_req(subsystem_nqn=nqn, host_nqn=host_nqn)
+                    )
+                    if add_host_response.status != 0:
+                        return add_host_response
+
+            return subsystem_response
```

Comment:
Nothing technically wrong with this but... (yeah there are BUTs :P ), as this is a "machine"/scriptable API (not directly meant for users), we wanted to keep it finegrained: "Do one single thing well". By creating a subsystem plus adding allowlisted hosts, we're breaking that rule.

We could have chosen instead the approach of having a single "God-object-like" REST Resource (Subsystems... and users could create Subsystem + listener + host allowlist + namespaces all together!). But we went the other way around: fine-grained API, mapping almost 1:1 to gRPC API.

Is there any reason to do that? From the UI it'd be just a matter of sending 2 REST requests.

Reply:
I wanted to create a UI api for this same reason, but @nizamial09 was okay so did in same api.
Yes with UI there are caveats: I could send two request from same form then need to manage when one fails other succeeds, also on deletion same sort of conflicts need to be checked. Overall it makes things on fromtend complex.

In past I tried managing multiple UI with same form and it was not a good idea from experience.
Either Ui form should be a wizard for that which is not proposed as of now.

I am okay of we create a UI API for this as that would off load most of work fro UI and keep the subsystem APi intact too

Reply:
combining this will be +1 from user experience POV. 
I think thats the advantage we want to provide users over CLI which has all of these as separate steps and commands already.

Reply:
I have revert changes, to use UI API, hence resolving.